firefox-ios/firefox-ios-tests/Tests/ClientTests/Telemetry/NotificationManagerTelemetryTests.swift

```diff
@@ -7,35 +7,59 @@ import XCTest
 
 @testable import Client
 
-// TODO: FXIOS-13744 - Migrate NotificationManagerTelemetryTests to use mock telemetry or GleanWrapper
 final class NotificationManagerTelemetryTests: XCTestCase {
+    var mockGleanWrapper: MockGleanWrapper!
+    override func setUp() {
+        super.setUp()
+        mockGleanWrapper = MockGleanWrapper()
+    }
+    override func tearDown() {
+        mockGleanWrapper = nil
+        super.tearDown()
+    }
+
     func test_onboardingNotificationPermission_GleanIsCalled() throws {
+        let event = GleanMetrics.Onboarding.notificationPermissionPrompt
+        typealias EventExtrasType = GleanMetrics.Onboarding.NotificationPermissionPromptExtra
         let isGranted = true
         let subject = createSubject()
-
         subject.sendNotificationPermissionPrompt(isPermissionGranted: isGranted)
-
-        try testEventMetricRecordingSuccess(metric: GleanMetrics.Onboarding.notificationPermissionPrompt)
+        let savedEvent = try XCTUnwrap(
+            mockGleanWrapper.savedEvents.first as? EventMetricType<EventExtrasType>
+        )
+        let savedExtras = try XCTUnwrap(
+            mockGleanWrapper.savedExtras.first as? EventExtrasType
+        )
+        XCTAssertEqual(mockGleanWrapper.recordEventCalled, 1)
+        XCTAssert(savedEvent === event, "Received \(savedEvent) instead of \(event)")
+        XCTAssertEqual(savedExtras.granted!, true)
     }
-
     func test_appNotificationPermission_GleanIsCalled() throws {
+        let event = GleanMetrics.App.notificationPermission
+        typealias EventExtrasType = GleanMetrics.App.NotificationPermissionExtra
         let subject = createSubject()
         let settings = MockUNNotificationSettings(authorizationStatus: .authorized, alertSetting: .enabled)
-
         subject.sendNotificationPermission(settings: settings)
-
-        try testEventMetricRecordingSuccess(metric: GleanMetrics.App.notificationPermission)
+        let savedEvent = try XCTUnwrap(
+            mockGleanWrapper.savedEvents.first as? EventMetricType<EventExtrasType>
+        )
+        let savedExtras = try XCTUnwrap(
+            mockGleanWrapper.savedExtras.first as? EventExtrasType
+        )
+        XCTAssertEqual(mockGleanWrapper.recordEventCalled, 1)
+        XCTAssertEqual(savedExtras.alertSetting!, "enabled", "Expected alertSetting string to be 'enabled'")
+        XCTAssertEqual(savedExtras.status!, "authorized", "Expected status string to be 'authorized'")
```

Comment:
nit: do we need the forced unwraps here? If not, let's remove that! Also to be consistent with how we test `savedExtras`, we can remove the message since we can infer that the string is not correct when the following fails without the comment.

```suggestion
        XCTAssertEqual(savedExtras.alertSetting, "enabled")
        XCTAssertEqual(savedExtras.status, "authorized")
```

Reply:
Hi again, i just updated my PR